packages/mobile/src/hooks/useUpdateRequired.ts

```diff
@@ -0,0 +1,41 @@
+import { useMemo } from 'react'
+
+import { StringKeys } from 'audius-client/src/common/services/remote-config'
+import VersionNumber from 'react-native-version-number'
+
+import { useRemoteVar } from 'app/hooks/useRemoteConfig'
+
+const isUpdateNeeded = (version: string, minVersion: string): boolean => {
+  const parts = version.split('.')
+  const minParts = minVersion.split('.')
+  const versionInfo = {
+    major: Number(parts[0]),
+    minor: Number(parts[1]),
+    patch: Number(parts[2])
+  }
+  const minVersionInfo = {
+    major: Number(minParts[0]),
+    minor: Number(minParts[1]),
+    patch: Number(minParts[2])
+  }
+
+  if (versionInfo.major > minVersionInfo.major) return false
+  if (versionInfo.major < minVersionInfo.major) return true
+
+  if (versionInfo.minor > minVersionInfo.minor) return false
+  if (versionInfo.minor < minVersionInfo.minor) return true
+
+  return versionInfo.patch < minVersionInfo.patch
+}
+
+export const useUpdateRequired = () => {
+  const { appVersion } = VersionNumber
+  const minAppVersion = useRemoteVar(StringKeys.MIN_APP_VERSION)
+
+  const updateRequired = useMemo(
+    () => isUpdateNeeded(appVersion, minAppVersion),
```

Comment:
i think this will return true or false right, so it's not necessarily needed to memoize? is it just to avoid the computation?